File: src/main/java/com/github/eendroroy/sdk/openweather/response/WeatherResponse.java
package com.github.eendroroy.sdk.openweather.response;

import com.fasterxml.jackson.annotation.JsonProperty;
import com.fasterxml.jackson.annotation.JsonRootName;
import lombok.Getter;
import lombok.Setter;
import lombok.ToString;

/**
 * @author indrajit
 */

@ToString
@Setter
@Getter
@JsonRootName("weather")
public class WeatherResponse implements BaseResponse {
    @JsonProperty("id")
    public Integer id;

    @JsonProperty("main")
    public String main;

    @JsonProperty("description")
    public String description;

    @JsonProperty("icon")
    public String icon;
}

File: src/main/kotlin/com/github/eendroroy/sdk/openweathermap/response/WeatherResponse.kt
package com.github.eendroroy.sdk.openweathermap.response

import com.fasterxml.jackson.annotation.JsonProperty
import com.fasterxml.jackson.annotation.JsonRootName

/**
 * @author indrajit
 */
@JsonRootName("weather")
class WeatherResponse : BaseResponse {
    @JsonProperty("id")
    var id: Int? = null

    @JsonProperty("main")
    var main: String? = null

    @JsonProperty("description")
    var description: String? = null

    @JsonProperty("icon")
    var icon: String? = null

    @JsonProperty("icon_url")
    fun iconUrl(): String {
        return "http://openweathermap.org/img/wn/$icon@2x.png"
    }
}